subsys/net/lib/nrf_cloud/src/nrf_cloud_fota_support.c

```diff
@@ -0,0 +1,554 @@
+
+/****************************************************/
+/* Define wrappers for transport-specific functions */
+/****************************************************/
+
+static bool check_ctx(struct nrf_cloud_fota_support_ctx *ctx)
+{
+	if (!ctx || !ctx->reboot_fn) {
+		return false;
+	}
+	if (IS_ENABLED(CONFIG_NRF_CLOUD_REST) && (!ctx->rest_ctx || !ctx->device_id)) {
+		return false;
+	}
+	return true;
+}
+
+static int fota_job_get(struct nrf_cloud_fota_support_ctx *ctx,
+		 struct nrf_cloud_fota_job_info *const job)
+{
+#if defined(CONFIG_NRF_CLOUD_REST)
+	return nrf_cloud_rest_fota_job_get(ctx->rest_ctx, ctx->device_id, job);
+#elif defined(CONFIG_NRF_CLOUD_COAP)
+	return nrf_cloud_coap_fota_job_get(job);
+#endif
+	return -ENOTSUP;
+}
+
+static int fota_job_update(struct nrf_cloud_fota_support_ctx *ctx, const char *const job_id,
+			   const enum nrf_cloud_fota_status status, const char * const details)
+{
+#if defined(CONFIG_NRF_CLOUD_REST)
+	return nrf_cloud_rest_fota_job_update(ctx->rest_ctx, ctx->device_id, job_id,
+					      status, details);
+#elif defined(CONFIG_NRF_CLOUD_COAP)
+	return nrf_cloud_coap_fota_job_update(job_id, status, details);
+#endif
+	return -ENOTSUP;
+}
+
+static int disconnect(struct nrf_cloud_fota_support_ctx *ctx)
+{
+	int err;
+
+#if defined(CONFIG_NRF_CLOUD_REST)
+	err = nrf_cloud_rest_disconnect(ctx->rest_ctx);
+#elif defined(CONFIG_NRF_CLOUD_COAP)
+	err = nrf_cloud_coap_disconnect();
+#else
+	err = -ENOTSUP;
+#endif
+#if defined(CONFIG_LTE_LINK_CONTROL)
+	(void)lte_lc_deinit();
+#endif
+
+	return err;
+}
+
+static void cleanup(void)
+{
+#if defined(CONFIG_NRF_CLOUD_REST)
+	nrf_cloud_rest_fota_job_free(&job);
+#elif defined(CONFIG_NRF_CLOUD_COAP)
+	nrf_cloud_coap_fota_job_free(&job);
+#endif
+}
+
+/****************************************************/
+/* End of transport-specific wrappers.              */
+/****************************************************/
+
+static int fota_settings_set(const char *key, size_t len_rd, settings_read_cb read_cb,
+				  void *cb_arg)
+{
+	ssize_t sz;
+
+	if (!key) {
+		LOG_DBG("Key is NULL");
+		return -EINVAL;
+	}
+
+	LOG_DBG("Settings key: %s, size: %d", key, len_rd);
+
+	if (strncmp(key, FOTA_SETTINGS_KEY_PENDING_JOB, strlen(FOTA_SETTINGS_KEY_PENDING_JOB))) {
+		return -ENOMSG;
+	}
+
+	if (len_rd > sizeof(pending_job)) {
+		LOG_INF("FOTA settings size larger than expected");
+		len_rd = sizeof(pending_job);
+	}
+
+	sz = read_cb(cb_arg, (void *)&pending_job, len_rd);
+	if (sz == 0) {
+		LOG_DBG("FOTA settings key-value pair has been deleted");
+		return -EIDRM;
+	} else if (sz < 0) {
+		LOG_ERR("FOTA settings read error: %d", sz);
+		return -EIO;
+	}
+
+	if (sz == sizeof(pending_job)) {
+		LOG_INF("Saved job: %s, type: %d, validate: %d, bl: 0x%X",
+			pending_job.id, pending_job.type,
+			pending_job.validate, pending_job.bl_flags);
+	} else {
+		LOG_INF("FOTA settings size smaller than current, likely outdated");
+	}
+
+	return 0;
+}
```

Comment:
This is duplicate code with the implementation of `fota_settings_set` in `nrf_cloud_fota.c` right?

It might be nice to merge this with that over there. But it doesn't have to be for this PR. But if not this PR, can we write a ticket about it?

Reply:
ditto for other job storage functions

Maybe we need `nrf_cloud_fota_job.c` or something like that, with an internal private header. Again though, it's fine by me if you think that should be a followup PR

Reply:
I like the idea of moving it to `nrf_cloud_fota_job.c`.  I'm not certain I want to do it now, but maybe before the end of the sprint.

Reply:
I decided to consolidate the settings handling in nrf_cloud_fota_common.c, but leave the rest as it is for now.